docs/llm_tools.md

```diff
@@ -0,0 +1,204 @@
+| `context`         | Context | The `homeassistant.core.Context` of the conversation                                                    |
+| `user_prompt`     | string  | The raw text input that initiated the tool call                                                         |
+| `language`        | string  | The language of the conversation agent, or "*" for any language                                         |
+| `agent_id`        | string  | The ID of the conversation agent                                                                        |
+| `conversation_id` | string  | The ID of the conversation                                                                              |
+| `device_id`       | string  | The device ID that user uses for the conversation, if available                                         |
+| `assistant`       | string  | The assistant name used to control exposed entities. Currently only `conversation` is supported.        |
+
+### Unregistration
+When your integration no longer wants to provide the tool, you need to unregister it with `llm.async_remove_tool`.
+
+Example:
+```python
+async_remove_tool(hass, MyTool)
+```
+
+You can also use the tool name:
+```python
+async_remove_tool(hass, "multiply")
+```
+
+### Full example
+
+The following tool returns a currency exponent.
+
+```python
+from typing import Any
+
+from iso4217 import Currency
+import voluptuous as vol
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import llm
+
+
+def currency_validator(currency_code: str) -> Currency:
+    """Validate currency code."""
+    return Currency(currency_code)
+
+
+class MyTool(llm.Tool):
+    """LLM Tool to get currency exponent."""
+
+    name = "currency_exponent"
+    description = "Return the currency exponent for a ISO 4217 currency code"
+    parameters = vol.Schema(
+        {
+            vol.Required(
+                "currency", description="ISO 4217 alphabetic code"
+            ): currency_validator,
+        }
+    )
+
+    async def async_call(self, hass: HomeAssistant, tool_input: llm.ToolInput) -> Any:
+        """Call the function."""
+        return {"exponent": tool_input.tool_args["currency"].exponent}
+
+
+llm.async_register_tool(hass, MyTool())
+```
+
+## Option 3. Tool function
+
+Define a function and register it as an LLM tool. This is the easiest method.
+
+There are two ways to register the function as an LLM tool: with `llm.FunctionTool` class or with a decorator.
+
+Example with `llm.FunctionTool` class:
+
+```python
+from homeassistant.helpers import llm
+
+
+def multiply(a: int, b: int) -> dict[str, int]:
+    "Return a product of two integers."
+    return {"result": a * b}
+
+
+my_tool = llm.FunctionTool(multiply)
+llm.async_register_tool(hass, my_tool)
+```
+
+Example with the `@llm_tool` decorator:
+```python
+from homeassistant.helpers.llm import llm_tool
+
+
+@llm_tool(hass)
```

Comment:
This example doesn't really show where `hass` comes from and given its only useful for inline functions where `hass` came from some other function, i'm not sure its adding value as a decorator over the previous example.